### ISSUE TITLE
1.3.3 install broken, ModuleNotFoundError: No module named 'pkgconfig'

### ISSUE THREAD

**Issue by gs202** [mehcode/python-xmlsec#108]
During installation i'm getting:
`
ModuleNotFoundError: No module named 'pkgconfig'
`
But as can be seen at the stack trace: 
`
Installing pkgconfig-1.5.1-py2.py3-none-any.whl to /opt/atlassian/pipelines/agent/build/.eggs
Installed /opt/atlassian/pipelines/agent/build/.eggs/pkgconfig-1.5.1-py3.6.egg
`

both packages are in my requirements file
`
xmlsec==1.3.3             # via python3-saml
pkgconfig==1.5.1          # via xmlsec
`

```
python3.6 setup.py test --addopts "-m unit_test"
running pytest
Searching for zipp==0.3.3
Reading https://pypi.org/simple/zipp/
Downloading https://files.pythonhosted.org/packages/00/42/10a1f8d29eaf9c7f2b555a5d1d602b9261f2a0cbec08f8f23d9056f5915d/zipp-0.3.3-py2.py3-none-any.whl#sha256=682b3e1c62b7026afe24eadf6be579fb45fec54c07ea218bded8092af07a68c4
Best match: zipp 0.3.3
Processing zipp-0.3.3-py2.py3-none-any.whl
Installing zipp-0.3.3-py2.py3-none-any.whl to /opt/atlassian/pipelines/agent/build/.eggs
writing requirements to /opt/atlassian/pipelines/agent/build/.eggs/zipp-0.3.3-py3.6.egg/EGG-INFO/requires.txt
Installed /opt/atlassian/pipelines/agent/build/.eggs/zipp-0.3.3-py3.6.egg
Searching for pkgconfig==1.5.1
Reading https://pypi.org/simple/pkgconfig/
Downloading https://files.pythonhosted.org/packages/b4/2c/bf434cb5a6590417e1d4468050ec317ea17fd6231c2a256df4646c11e588/pkgconfig-1.5.1-py2.py3-none-any.whl#sha256=cddf2d7ecadb272178a942eb852a9dee46bda2adcc36c3416b0fef47a4ed9f38
Best match: pkgconfig 1.5.1
Processing pkgconfig-1.5.1-py2.py3-none-any.whl
Installing pkgconfig-1.5.1-py2.py3-none-any.whl to /opt/atlassian/pipelines/agent/build/.eggs
Installed /opt/atlassian/pipelines/agent/build/.eggs/pkgconfig-1.5.1-py3.6.egg
Searching for xunitparser==1.3.3
Reading https://pypi.org/simple/xunitparser/
Downloading https://files.pythonhosted.org/packages/57/04/15064f68aea845109382e966ea059497bcbf006b2ef0dee0e602065e09ce/xunitparser-1.3.3.tar.gz#sha256=f6e9262bbaeb9be8fccd52f1fd18a3f9b09aa95933df18137abd82197db25515
Best match: xunitparser 1.3.3
Processing xunitparser-1.3.3.tar.gz
Writing /tmp/easy_install-9jgca8r8/xunitparser-1.3.3/setup.cfg
Running xunitparser-1.3.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-9jgca8r8/xunitparser-1.3.3/egg-dist-tmp-k3cs_bhd
zip_safe flag not set; analyzing archive contents...
Moving xunitparser-1.3.3-py3.6.egg to /opt/atlassian/pipelines/agent/build/.eggs
Installed /opt/atlassian/pipelines/agent/build/.eggs/xunitparser-1.3.3-py3.6.egg
Searching for xmlsec==1.3.3
Reading https://pypi.org/simple/xmlsec/
Downloading https://files.pythonhosted.org/packages/35/42/d7cd323c91d4706f3cc32ffe7d5f851ab8ef9898ccb350f6ba593dd8b89a/xmlsec-1.3.3.tar.gz#sha256=e573c0172174973223d874ffd158ecd4e0faa761015474385289a6468dd29ed6
Best match: xmlsec 1.3.3
Processing xmlsec-1.3.3.tar.gz
Writing /tmp/easy_install-3psrchsy/xmlsec-1.3.3/setup.cfg
Running xmlsec-1.3.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-3psrchsy/xmlsec-1.3.3/egg-dist-tmp-iyhjizxz
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/setup.py", line 60, in <module>
  File "/usr/local/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/usr/local/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/local/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/local/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "/usr/local/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/local/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/usr/local/lib/python3.6/distutils/command/install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "/usr/local/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/local/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/setup.py", line 11, in run
    author='orcarmi',
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/setup.py", line 16, in patch_options
    )
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/xmlsec_setupinfo.py", line 48, in define_macros
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/xmlsec_setupinfo.py", line 111, in xmlsec_config
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/xmlsec_setupinfo.py", line 164, in load_xmlsec1_config
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/.eggs/pkgconfig-1.5.1-py3.6.egg/pkgconfig/pkgconfig.py", line 248, in parse
  File "/tmp/easy_install-3psrchsy/xmlsec-1.3.3/.eggs/pkgconfig-1.5.1-py3.6.egg/pkgconfig/pkgconfig.py", line 103, in _raise_if_not_exists
pkgconfig.pkgconfig.PackageNotFoundError: xmlsec1 not found
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "setup.py", line 15, in <module>
    tests_require=Path('requirements.txt').read_text(),
  File "/usr/local/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/usr/local/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/local/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/local/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/opt/atlassian/pipelines/agent/build/.eggs/pytest_runner-4.4-py3.6.egg/ptr.py", line 183, in run
    installed_dists = self.install_dists(dist)
  File "/opt/atlassian/pipelines/agent/build/.eggs/pytest_runner-4.4-py3.6.egg/ptr.py", line 131, in install_dists
    orig.test.install_dists(dist),
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/test.py", line 206, in install_dists
    tr_d = dist.fetch_build_eggs(dist.tests_require or [])
  File "/usr/local/lib/python3.6/site-packages/setuptools/dist.py", line 514, in fetch_build_eggs
    replace_conflicting=True,
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 770, in resolve
    replace_conflicting=replace_conflicting
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1053, in best_match
    return self.obtain(req, installer)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1065, in obtain
    return installer(requirement)
  File "/opt/atlassian/pipelines/agent/build/.eggs/pytest_runner-4.4-py3.6.egg/ptr.py", line 72, in fetch_build_egg
    return cmd.easy_install(req)
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 673, in easy_install
    return self.install_item(spec, dist.location, tmpdir, deps)
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 699, in install_item
    dists = self.install_eggs(spec, download, tmpdir)
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 884, in install_eggs
    return self.build_and_install(setup_script, setup_base)
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 1152, in build_and_install
    self.run_setup(setup_script, setup_base, args)
  File "/usr/local/lib/python3.6/site-packages/setuptools/command/easy_install.py", line 1138, in run_setup
    run_setup(setup_script, args)
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 253, in run_setup
    raise
  File "/usr/local/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/usr/local/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 166, in save_modules
    saved_exc.resume()
  File "/usr/local/lib/python3.6/site-packages/setuptools/sandbox.py", line 140, in resume
    type, exc = map(pickle.loads, self._saved)
ModuleNotFoundError: No module named 'pkgconfig'
```

**Comment by anoasis** [comment]
Did you try yum install pkgconfig?
If you did, you probably need to configure PKG_CONFIG_PATH as well.

**Comment by hoefling** [comment]
@gs202 this was caused by `pkgconfig` not been declared as build dependency. This is now fixed in the latest 1.3.8 version; can you please confirm the issue is resolved?

**Comment by hoefling** [comment]
Closed as cannot reproduce this. Feel free to ping me if the issue should be reopened, though.
